src/Frame.cpp

```diff
@@ -149,7 +149,7 @@ Frame::Frame(const Frame& rhs) :
   F_(0),
   remd_indices_(rhs.remd_indices_),
   Mass_(rhs.Mass_),
-  memIsExternal_(rhs.memIsExternal_)
+  memIsExternal_(false)
```

Comment:
This feels correct - a copy constructor should indeed make a copy. However you may want to benchmark and see if there are any places where this hurts performance. There may be cases where we want to just pass the address of `X_` etc - in such cases we can create a new function to do that. Also, can you get rid of the `if (memIsExternal_)` part? If `memIsExternal_` is always false then it will never get called.